frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -512,10 +514,36 @@ class NativeQueryEditor extends Component {
     });
   };
 
+  togglePromptVisibility = () => {
+    this.setState(prev => ({
+      isPromptInputVisible: !prev.isPromptInputVisible,
+    }));
+  };
+
+  handleQueryUpdate = queryText => {
+    this._editor.setValue(queryText);
+    this._editor.clearSelection();
+  };
+
+  handleQueryGenerated = queryText => {
+    this.handleQueryUpdate(queryText);
+    this._editor.focus();
+  };
+
+  isPromptInputVisible = () => {
+    const { canUsePromptInput } = this.props;
+    const database = this.props.query.database();
+    const isSupported =
```

Comment:
I believe we should do something similar in metabot components. Could we create a common function for this?